TEST_APPS/device/nfcapp/nfccommands.cpp

```
@@ -0,0 +1,203 @@
+#include "rtos\Thread.h"
+
+#include "nfctestshim.h"
+#include "nfcprocess.h"
+#include "nfccommands.h"
+
+events::EventQueue nfcQueue;
+Thread nfcThread;
+NFCTestShim * pNFC_Test_Shim = NULL;
+
+
+
+NFCTestShim* new_testshim() {
+#if MBED_CONF_NFCEEPROM
+    mbed::nfc::NFCEEPROMDriver& eeprom_driver = get_eeprom_driver(nfcQueue);
+
+    return ( (NFCTestShim *)(new NFCProcessEEPROM(nfcQueue, eeprom_driver)) );
+#else
+    return( (NFCTestShim *)(new NFCProcessController(nfcQueue)) );
+#endif // EEPROM
+
+}
+
+void nfcRoutine() {
+    nfcQueue.dispatch_forever();
+}
+
+HandleTestCommand::HandleTestCommand() {
+    osStatus status = nfcThread.start(callback(&nfcRoutine) );
+    MBED_ASSERT(status == osOK);
+}
+
+int HandleTestCommand::cmd_set_last_nfc_error(int argc, char *argv[]){
+    if (argc <= 1){
+        cmd_printf("setlastnfcerror() invalid parameter(s)\r\n");
+        return(CMDLINE_RETCODE_INVALID_PARAMETERS);
+    } else {
+        int value = strtol(argv[1], NULL, 10);
+        nfcQueue.call(NFCTestShim::on_set_last_nfc_error, value);
+    }
+    return(CMDLINE_RETCODE_EXCUTING_CONTINUE);
+}
+
+
+int HandleTestCommand::cmd_init_nfc(int argc, char *argv[]){
+
+	if (pNFC_Test_Shim) {
+		cmd_printf("WARN init called again!\r\n"); // only legal here, if eeprom driver stops talking
+	} else {
+	    pNFC_Test_Shim = new_testshim();
+	}
+	nfcQueue.call(pNFC_Test_Shim, &NFCTestShim::on_init);
+	return(CMDLINE_RETCODE_EXCUTING_CONTINUE);
+}
+
+int HandleTestCommand::cmd_read_message(int argc, char *argv[]) {
+    nfcQueue.call(pNFC_Test_Shim, &NFCTestShim::on_read_nfceeprom);
+
+    return(CMDLINE_RETCODE_EXCUTING_CONTINUE);
+}
+
+int HandleTestCommand::cmd_set_smartposter(int argc, char *argv[]) {
+    if (argc <= 1){
+        cmd_printf("setlastnfcerror() invalid parameter(s)\r\n");
+        return(CMDLINE_RETCODE_INVALID_PARAMETERS);
+    } else {
+        // parse arg and queue it up
+        char * uri = (char*)malloc(MAX_URL_LENGTH+1);
+        if (uri) {
```

Should return an error (out of memory) if this is NULL